dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/map-fee-record-payment-entity-groups-to-fee-record-payment-groups.test.ts

```diff
@@ -177,41 +274,135 @@ describe('get-utilisation-report-reconciliation-details-by-id.controller helpers
+        ];
+        const group: FeeRecordPaymentEntityGroup = {
+          feeRecords,
+          payments: [PaymentEntityMockBuilder.forCurrency('GBP').build()],
+        };
+
+        // Act
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups([group]);
+
+        // Assert
+        expect(result).toHaveLength(1);
+        expect(result[0].dateReconciled).toBeNull();
+        expect(result[0].reconciledByUser).toBeNull();
+      });
+
+      it('sets the dateReconciled and reconciledByUser to the most recently reconciled fee record when all the fee records in the group are at the RECONCILED status', async () => {
+        // Arrange
+        const feeRecords = [
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'abc123', dateReconciled: new Date('2023') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'def456', dateReconciled: new Date('2024') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'ghi789', dateReconciled: new Date('2022') }).build(),
+        ];
+        const group: FeeRecordPaymentEntityGroup = {
+          feeRecords,
+          payments: [PaymentEntityMockBuilder.forCurrency('GBP').build()],
+        };
+
+        when(findTfmUserSpy)
+          .calledWith('def456')
+          .mockResolvedValue({ ...aTfmUser(), firstName: 'John', lastName: 'Smith' });
+
+        // Act
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups([group]);
+
+        // Assert
+        expect(result).toHaveLength(1);
+        expect(result[0].reconciledByUser).toEqual<FeeRecordReconciledByUser>({ firstName: 'John', lastName: 'Smith' });
+        expect(result[0].dateReconciled).toEqual(new Date('2024'));
+      });
+
+      it('sets the dateReconciled to the most recently reconciled fee record and the reconciledByUser to null when all the fee records in the group are at the RECONCILED status but the latest fee record has no defined user', async () => {
+        // Arrange
+        const feeRecords = [
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'abc123', dateReconciled: new Date('2023') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'def456', dateReconciled: new Date('2024') }).build(),
+          reconciledFeeRecordBuilder({ reconciledByUserId: 'ghi789', dateReconciled: new Date('2022') }).build(),
+        ];
+        const group: FeeRecordPaymentEntityGroup = {
+          feeRecords,
+          payments: [PaymentEntityMockBuilder.forCurrency('GBP').build()],
+        };
+
+        when(findTfmUserSpy).calledWith('def456').mockResolvedValue(null);
+
+        // Act
+        const result = await mapFeeRecordPaymentEntityGroupsToFeeRecordPaymentGroups([group]);
+
+        // Assert
+        expect(result).toHaveLength(1);
+        expect(result[0].reconciledByUser).toBeNull();
+        expect(result[0].dateReconciled).toEqual(new Date('2024'));
+      });
```

Comment:
Is there anyway we could split this test up?

Reply:
I'd prefer to keep these assertions together as these two fields are quite tightly coupled in terms of when they get populated, so it makes sense contextually to test them as a pair